src/main/java/sg/ncl/MainController.java

```diff
@@ -3341,6 +3347,162 @@ public String adminEnergy(Model model,
         return "energy_usage";
     }
 
+    /**
+     * Allows admins to:
+     * view reservations
+     * reserve nodes
+     * release nodes
+     */
+    @GetMapping("/admin/nodesReservation")
+    public String adminNodesReservation(@ModelAttribute("reservationStatusForm") ReservationStatusForm reservationStatusForm, Model model, HttpSession session) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        model.addAttribute(ALL_TEAMS, getTeamMap());
+        model.addAttribute(RESERVATION_STATUS_FORM, reservationStatusForm);
+
+        return "node_reservation";
+    }
+
+    @PostMapping("/admin/nodesReservation")
+    public String adminNodesReservation(@ModelAttribute("reservationStatusForm") ReservationStatusForm reservationStatusForm, BindingResult bindingResult, RedirectAttributes redirectAttributes, Model model) throws IOException {
+        HashMap<String, Team2> teamMap = getTeamMap();
+
+        // sanitization
+        if (bindingResult.hasErrors()) {
+            model.addAttribute(ALL_TEAMS, teamMap);
+            model.addAttribute(RESERVATION_STATUS_FORM, reservationStatusForm);
+            model.addAttribute(STATUS, NODES_RESERVATION_FAIL);
+            model.addAttribute(MESSAGE, "form errors");
+            return "node_reservation";
+        }
+
+        switch(reservationStatusForm.getAction()) {
+            case "release":
+                releaseNodes(reservationStatusForm, redirectAttributes);
+                break;
+            case "reserve":
+                reserveNodes(reservationStatusForm, redirectAttributes);
+                break;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)